pyat/at/physics/rdt.py

```diff
@@ -0,0 +1,569 @@
+            second_order,
+            pf,
+        )
+        rdtlist.append(rdt)
+        rdtlist2.append(rdt2)
+    return rdtlist, rdtlist2
+
+
+def get_rdts(
+    ring: Lattice,
+    refpts: Refpts,
+    rdt_type: Sequence[RDTType] | RDTType,
+    second_order: bool = False,
+    use_mp: bool = False,
+    pool_size: int = None,
+):
+    """
+    :py:func:`get_rdts` computes the ring RDTs based on the original implementation
+    from ELEGANT. For consistency, pyAT keeps the sign convention of the AT MATLAB interface
+
+    Refs [1]_ and [2]_ were used to calculate the magnitude of first and second order rdts and
+    Ref. [3]_ for the focusing rdt, however, different sign conventions are used along the references
+    and in the original implementation. To overcome this issue, first and second order rdts
+    are provided as a total and also separately so the user can redefine these conventions if needed.
+
+    The resonance base of GEOMETRIC2 rdts is explained in Eq. (54) of Ref [4]_.
+
+    The periodicity property of the lattice is automatically taken into account in the rdt
+    calculation, however the calculation of the second order contribution of sextupoles
+    to the GEOMETRIC2 and DETUNING RDT types can only be derived for periodicity=1
+    (i.e. full ring provided)
+
+    Usage:
+      >>> get_rdts(ring, reftps, [RDTType.COUPLING, RDTType.CHROMATIC])
+
+    Parameters:
+        ring: :code:`at.Lattice` object
+        refpts: Element refpts at which the RDTs are calculated
+        rdt_type: Type of RDTs to be calculated. The type can be
+        :code:`Sequence[at.RDTType] | at.RDTType`.
+        Possible RDT types are:
+            at.RDTType.ALL: all available RDTs
+            at.RDTType.FOCUSING: Normal quadrupole RDTs
+            at.RDTType.COUPLING: Linear coupling RDTs from skew quadrupoles
+            at.RDTType.CHROMATIC: Chromatic RDTs from sextupoles and normal quadrupoles
+            at.RDTType.GEOMETRIC1: Geometric RDTs from sextupoles
+            at.RDTType.GEOMETRIC2: Geometric RDTs from octupoles
+              the second order contribution of sextupoles is added when
+              :code:`second_order = True`
+            at.RDTType.TUNESHIFT: Amplitude detuning coefficients
+              the second order contribution of sextupoles is added when
+              :code:`second_order = True`
+
+    Keyword Args:
+        second_order: Compute second order terms (default: False).
+          Computation is significantly longer using this method
+        use_mp: Activate parallel calculation
+        pool_size: Number of processes used for parallelization
+
+    Returns:
+        rdts: rdt data (complex) at refpts
+        rdts2: (complex) contribution from sextupole second order terms
+          Available only for GEOMETRIC2 terms
+        rdttot: (complex) total rdts
+
+        **rdts** is a dictionary with keys:
+        =================   ======
+        **refpts**           location of the rdt
```

Comment:
There is one extra space before location that does not match the column alignment.

Reply:
Dear @swhite2401, all other test are  correct. After fixing this typo you could merge.